File: android/SalmonTrollingAssistant/app/src/main/java/com/example/salmontrollingassistant/domain/service/AuthenticationService.kt
package com.example.salmontrollingassistant.domain.service

import com.example.salmontrollingassistant.domain.model.AuthCredentials
import com.example.salmontrollingassistant.domain.model.AuthResult
import com.example.salmontrollingassistant.domain.model.UserProfile
import kotlinx.coroutines.flow.Flow

interface AuthenticationService {
    /**
     * Create a new user account with email and password
     */
    suspend fun createAccount(credentials: AuthCredentials, name: String?): AuthResult
    
    /**
     * Sign in with existing credentials
     */
    suspend fun signIn(credentials: AuthCredentials): AuthResult
    
    /**
     * Sign out the current user
     */
    suspend fun signOut(): Boolean
    
    /**
     * Create or get an anonymous user account
     */
    suspend fun signInAnonymously(): AuthResult
    
    /**
     * Convert an anonymous account to a permanent account
     */
    suspend fun convertAnonymousAccount(credentials: AuthCredentials, name: String?): AuthResult
    
    /**
     * Get the current user profile
     */
    fun getCurrentUser(): Flow<UserProfile?>
    
    /**
     * Check if the user is signed in
     */
    fun isSignedIn(): Flow<Boolean>
    
    /**
     * Check if the current user is anonymous
     */
    fun isAnonymousUser(): Flow<Boolean>
}

File: ios/SalmonTrollingAssistant/SalmonTrollingAssistant/Domain/Services/AuthenticationService.swift
import Foundation
import Combine

protocol AuthenticationService {
    /**
     * Create a new user account with email and password
     */
    func createAccount(credentials: AuthCredentials, name: String?) async -> AuthResult
    
    /**
     * Sign in with existing credentials
     */
    func signIn(credentials: AuthCredentials) async -> AuthResult
    
    /**
     * Sign out the current user
     */
    func signOut() async -> Bool
    
    /**
     * Create or get an anonymous user account
     */
    func signInAnonymously() async -> AuthResult
    
    /**
     * Convert an anonymous account to a permanent account
     */
    func convertAnonymousAccount(credentials: AuthCredentials, name: String?) async -> AuthResult
    
    /**
     * Get the current user profile
     */
    var currentUser: CurrentValueSubject<UserProfile?, Never> { get }
    
    /**
     * Check if the user is signed in
     */
    var isSignedIn: AnyPublisher<Bool, Never> { get }
    
    /**
     * Check if the current user is anonymous
     */
    var isAnonymousUser: AnyPublisher<Bool, Never> { get }
}